experiments/functional/backup_and_restore/setup_dependency.yml

```diff
@@ -18,47 +18,86 @@
+     register: minio_status
+     until: "'Running' in minio_status.stdout"
+     delay: 5
+     retries: 15
+   
+   - name: Waiting for minio job to create bucket
+     shell: kubectl get pod -n velero -l job-name=minio-setup -o jsonpath='{.items[*].status.phase}'
+     register: minio_job_status
+     until: "'Succeeded' in minio_job_status.stdout"
+     delay: 5
+     retries: 10
+
+  when: bucket_type == "minio"
 
 - block:
 
-   - name: Installing velero server inside cluster
-     shell: >
-        velero install \
-          --provider aws \
-          --bucket velero \
-          --secret-file ./credentials-velero \
-          --use-volume-snapshots=false \
-          --use-restic \
-          --backup-location-config region=minio,s3ForcePathStyle="true",s3Url=http://minio.velero.svc:9000
-     when: velero_version == "v1.1.0"
-
-   - name: Installing velero server inside cluster
-     shell: >
-        velero install \
-          --provider aws \
-          --bucket velero \
-          --secret-file ./credentials-velero \
-          --plugins velero/velero-plugin-for-aws:v1.0.0 \
-          --use-volume-snapshots=false \
-          --use-restic \
-          --backup-location-config region=minio,s3ForcePathStyle="true",s3Url=http://minio.velero.svc:9000
-     when: velero_version == "v1.2.0"
+   - name: Fetching service account key.json from configmap
+     shell: kubectl get cm gcp-key -n litmus -o jsonpath='{.data.key\.json}' > ./key.json
+        
+   - name: Authenticating using gcp service account
+     shell: gcloud auth activate-service-account --key-file key.json
+
+   - name: Setting up the project ID
+     shell: gcloud config set project {{ gcp_project_id }} 
+
+   - name: Creating gcp bucket
+     shell: gsutil mb gs://e2e-gcp-bucket/
+
+  when: bucket_type == "GCP"
+
+- block:
+
+   - block:
+
+      - name: Installing velero server inside cluster
+        shell: >
+           velero install \
+             --provider aws \
+             --bucket velero \
+             --secret-file ./credentials-velero \
+             --use-volume-snapshots=false \
+             --use-restic \
+             --backup-location-config region=minio,s3ForcePathStyle="true",s3Url=http://minio.velero.svc:9000
+        when: velero_version == "v1.1.0"
+   
+      - name: Installing velero server inside cluster
+        shell: >
+           velero install \
+             --provider aws \
+             --bucket velero \
+             --secret-file ./credentials-velero \
+             --plugins velero/velero-plugin-for-aws:v1.0.0 \
+             --use-volume-snapshots=false \
+             --use-restic \
+             --backup-location-config region=minio,s3ForcePathStyle="true",s3Url=http://minio.velero.svc:9000
+        when: velero_version == "v1.2.0"
+
+     when: "bucket_type == 'minio'"
+
+   - block: 
+
+      - name: Installing velero server inside cluster
+        shell: >
+          velero install \
```

Comment:
In GCP packet is this velero server istallatoin is same for the version 1.1.0 and 1.2.0 ?

Reply:
Made the changes